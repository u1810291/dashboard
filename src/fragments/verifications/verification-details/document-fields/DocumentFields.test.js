import React from 'react'
import { shallow } from 'enzyme'
import DocumentFields from '.'

describe('DocumentFields', () => {
  test('Empty fields', () => {
    const component = shallow(<DocumentFields fields={[]} />)
    expect(component).toMatchSnapshot()
  })

  test('With editable fields', () => {
    const component = shallow(
      <DocumentFields
        fields={[
          {
            caption: 'Document field caption',
            value: 'Value',
            status: 'success',
            editable: true,
            id: 'FIELD_ID',
            docId: 'DOC_ID'
          }
        ]}
      />
    )
    expect(component).toMatchSnapshot()
  })

  test('With static fields', () => {
    const component = shallow(
      <DocumentFields
        fields={[
          {
            caption: 'Document field caption',
            value: 'Value',
            status: 'failure'
          }
        ]}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
