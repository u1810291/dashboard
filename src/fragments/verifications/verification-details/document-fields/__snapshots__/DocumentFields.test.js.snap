// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentFields Empty fields 1`] = `
<ul>
  <li
    className="text-secondary"
  >
    <FormattedMessage
      id="fragments.verifications.verification_detail.no_data"
      values={Object {}}
    />
  </li>
</ul>
`;

exports[`DocumentFields With editable fields 1`] = `
<ul>
  <li>
    <span
      className="text-secondary"
    >
      Document field caption
    </span>
    <img
      alt="success"
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "success.svg",
        }
      }
    />
    <span>
      <TextEditable
        onSubmit={[Function]}
        text="Value"
      />
    </span>
  </li>
</ul>
`;

exports[`DocumentFields With static fields 1`] = `
<ul>
  <li>
    <span
      className="text-secondary"
    >
      Document field caption
    </span>
    <img
      alt="failure"
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "failure.svg",
        }
      }
    />
    <span>
      Value
    </span>
  </li>
</ul>
`;
